backend/app/policies/enrollment_policy.rb

```diff
@@ -77,8 +77,8 @@ def archive?
       user.belongs_to_organization?(record) &&
       user.is_demandeur?(record)
 
-    instructor_right = user.is_instructor?(record.target_api)
-    administrator_right = user.is_administrator?
+    instructor_right = !record.status_validated? && user.is_instructor?(record.target_api)
+    administrator_right = !record.status_validated? && user.is_administrator?
```

Comment:
I don't think these conditions are useful as `record.can_archive_status?` returns already `false` if we're trying to transition from a `validated` state to an `archived` state.

Reply:
Indeed, `record.can_archive_status?` already check this behaviour.